### PR TITLE
Rename MapFeatureAssertionsSpec to MapAssertionsSpec

### PR DESCRIPTION
**Pull request by tarczynskitomek**
- Join existing Spec files after renaming
- First PR to solve #329 
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt [hunk 1]
```diff
@@ -81,14 +89,32 @@ abstract class MapAssertionsSpec(
             { containsKeyWithNullableValueAssertions(this, keyNullableValue("a") { }, arrayOf()) }
         )
     ) {})
+    include(object : SubjectLessSpec<Map<out String, Int>>(describePrefix,
```
Comment (robstoll) [inline misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt:92]: please fuse with line 55

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt [hunk 2]
```diff
@@ -81,14 +89,32 @@ abstract class MapAssertionsSpec(
...
+        keysFeature.forSubjectLess(),
+        keys.forSubjectLess { isEmpty() },
+        valuesFeature.forSubjectLess(),
+        values.forSubjectLess { isEmpty() },
+        getExistingFeature.forSubjectLess("a"),
+        getExisting.forSubjectLess("a") { isGreaterThan(1) }
+    ) {})
+    include(object : SubjectLessSpec<Map<out String?, Int?>>("$describePrefix[nullable] ",
```
Comment (robstoll) [inline misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt:100]: please fuse with line 67

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt [hunk 3]
```diff
@@ -81,14 +89,32 @@ abstract class MapAssertionsSpec(
...
+        getExistingNullableFeature.forSubjectLess("a"),
+        getExistingNullable.forSubjectLess("a") { toBe(null) }
+    ) {})
 
+    val map: Map<out String, Int> = mapOf("a" to 1, "b" to 2)
+    val fluent = expect(map)
+
+    include(object : AssertionCreatorSpec<Map<out String, Int>>(
```
Comment (robstoll) [inline misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt:108]: please fuse with 78

Comment (robstoll) [reply misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt:108]: It seems you did not move the following here:
```
    include(object : AssertionCreatorSpec<Map<String?, Int?>>(
        "$describePrefix[nullable] ", mapNullable,
        getExistingNullable.forAssertionCreatorSpec("$toBeDescr: 2", "b") { toBe(null) }
    ) {})
```
please fuse with 85

**Review by robstoll — CHANGES_REQUESTED** [review]

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/mapAssertions.kt [hunk 4]
```diff
@@ -142,7 +142,7 @@ val <V, T : Map<*, V>> Expect<T>.values: Expect<Collection<V>>
  * @return An [Expect] for the current subject of the assertion.
  * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
  */
-fun <K, V, T : Map<K, V>> Expect<T>.values(assertionCreator: Expect<Collection<V>>.() -> Unit): Expect<T> =
+fun <K, V, T : Map<out K, V>> Expect<T>.values(assertionCreator: Expect<Collection<V>>.() -> Unit): Expect<T> =
```
[the remark this answers is not in the document]
Comment (tarczynskitomek) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/mapAssertions.kt:145]: Thanks! Although Kotlin compiler helped me with this one ;-)

File: misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt (hunk 3 above)
Comment (tarczynskitomek) [reply misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/MapAssertionsSpec.kt:108]: Indeed, missed that one :(

File: apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/MapAssertionsSpec.kt [hunk 5]
```diff
@@ -277,5 +263,37 @@ class MapAssertionsSpec : ch.tutteli.atrium.specs.integration.MapAssertionsSpec(
         readOnlyNullableKeyValueMap = readOnlyNullableKeyValueMap notToBe empty
         starMap = starMap notToBe empty
 
-    }
+        map = map keys { }
```
Comment (robstoll) [inline apis/infix-en_GB/atrium-api-infix-en_GB-common/src/test/kotlin/ch/tutteli/atrium/api/infix/en_GB/MapAssertionsSpec.kt:266]: please also add `keys o` and `values o`

**Review by robstoll — CHANGES_REQUESTED** [review]
Thanks for the corrections. One tiny detail left and then we are all good.

**Review by robstoll — APPROVED** [review]

**Comment by robstoll** [comment]
@tarczynskitomek thanks for your first contribution to Atrium :tada:

**Pull request merged**